process/coordinator/blockDataRequest_test.go

```diff
@@ -154,6 +234,683 @@ func TestTransactionCoordinator_requestMissingMiniBlocksAndTransactionsShouldWor
+		args.PreProcessors = preprocContainer
+
+		blockDataRequester, _ := NewBlockDataRequester(args)
+
+		// Add the mini block to requested items
+		key := []byte("hash1")
+		_ = blockDataRequester.requestedItemsHandler.Add(string(key))
+
+		// Create mini block
+		miniBlock := &block.MiniBlock{
+			Type:     block.TxBlock,
+			TxHashes: [][]byte{[]byte("tx1"), []byte("tx2"), []byte("tx3")},
+		}
+
+		// Should process successfully
+		blockDataRequester.receivedMiniBlock(key, miniBlock)
+	})
+
+	t.Run("should handle mini block with no transactions", func(t *testing.T) {
+		t.Parallel()
+		args := createMockArgs()
+
+		// Create preprocessors container with mock
+		preprocContainer := containers.NewPreProcessorsContainer()
+		txPreproc := &preprocMocks.PreProcessorMock{
+			RequestTransactionsForMiniBlockCalled: func(miniBlock *block.MiniBlock) int {
+				return len(miniBlock.TxHashes)
+			},
+		}
+		_ = preprocContainer.Add(block.TxBlock, txPreproc)
+		args.PreProcessors = preprocContainer
+
+		blockDataRequester, _ := NewBlockDataRequester(args)
+
+		// Add the mini block to requested items
+		key := []byte("hash1")
+		_ = blockDataRequester.requestedItemsHandler.Add(string(key))
+
+		// Create mini block with no transactions
+		miniBlock := &block.MiniBlock{
+			Type:     block.TxBlock,
+			TxHashes: [][]byte{}, // Empty transactions
+		}
+
+		// Should process successfully (no transactions requested)
+		blockDataRequester.receivedMiniBlock(key, miniBlock)
+	})
+
+	t.Run("should handle multiple mini block types", func(t *testing.T) {
+		t.Parallel()
+		args := createMockArgs()
+
+		// Create preprocessors container with multiple mocks
+		preprocContainer := containers.NewPreProcessorsContainer()
+		txPreproc := &preprocMocks.PreProcessorMock{
+			RequestTransactionsForMiniBlockCalled: func(miniBlock *block.MiniBlock) int {
+				return len(miniBlock.TxHashes)
+			},
+		}
+		peerPreproc := &preprocMocks.PreProcessorMock{
+			RequestTransactionsForMiniBlockCalled: func(miniBlock *block.MiniBlock) int {
+				return len(miniBlock.TxHashes) * 2 // Different logic for peer blocks
+			},
+		}
+		_ = preprocContainer.Add(block.TxBlock, txPreproc)
+		_ = preprocContainer.Add(block.PeerBlock, peerPreproc)
+		args.PreProcessors = preprocContainer
+
+		blockDataRequester, _ := NewBlockDataRequester(args)
+
+		// Test TxBlock type
+		key1 := []byte("hash1")
+		_ = blockDataRequester.requestedItemsHandler.Add(string(key1))
+		txMiniBlock := &block.MiniBlock{
+			Type:     block.TxBlock,
+			TxHashes: [][]byte{[]byte("tx1"), []byte("tx2")},
+		}
+		blockDataRequester.receivedMiniBlock(key1, txMiniBlock)
+
+		// Test PeerBlock type
+		key2 := []byte("hash2")
+		_ = blockDataRequester.requestedItemsHandler.Add(string(key2))
+		peerMiniBlock := &block.MiniBlock{
+			Type:     block.PeerBlock,
+			TxHashes: [][]byte{[]byte("peer1"), []byte("peer2")},
+		}
+		blockDataRequester.receivedMiniBlock(key2, peerMiniBlock)
```

Comment:
does not check anything

should check that the required request functions were called for the expected miniblock types and txHashes.

Reply:
added extra checks